### PR TITLE
Add the OpenTelemetry schema URL to the InstrumentationLibraryInfo

### PR DESCRIPTION
**Pull request by jkwatson**
and to the corresponding API calls.

### CODE REVIEW

File: api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java [hunk 1]
```diff
@@ -126,6 +126,25 @@ public static Tracer getTracer(String instrumentationName, String instrumentatio
     return get().getTracer(instrumentationName, instrumentationVersion);
   }
 
+  /**
+   * Gets or creates a named and versioned tracer instance from the globally registered {@link
+   * TracerProvider}.
+   *
+   * <p>This is a shortcut method for {@code getTracerProvider().get(instrumentationName,
+   * instrumentationVersion)}
```
Comment (iNikem) [inline api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java:134]:
```suggestion
   * <p>This is a shortcut method for {@code getTracerProvider().get(instrumentationName,
   * instrumentationVersion, schemaUrl)}
```

File: exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/MetricAdapter.java [hunk 2]
```diff
@@ -58,12 +58,15 @@
           new ArrayList<>(entryResource.getValue().size());
       for (Map.Entry<InstrumentationLibraryInfo, List<Metric>> entryLibrary :
           entryResource.getValue().entrySet()) {
-        instrumentationLibraryMetrics.add(
+        InstrumentationLibraryMetrics.Builder metricsBuilder =
```
Comment (kubawach) [inline exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/MetricAdapter.java:61]: Small suggestion - with extra "if" in the builder flow, perhaps it makes sense to move the Builder creation/configuration to a separate (helper) method? Then parent "for" flow will be a slightly more readable (for (....) instrumentationLibraryMetrics.add(metricsBuilderFor(entryLibrary).build())

File: exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/SpanAdapter.java [hunk 3]
```diff
@@ -75,13 +75,17 @@
...
           librarySpans.forEach(
-              (library, spans) ->
-                  resourceSpansBuilder.addInstrumentationLibrarySpans(
-                      InstrumentationLibrarySpans.newBuilder()
-                          .setInstrumentationLibrary(
-                              CommonAdapter.toProtoInstrumentationLibrary(library))
-                          .addAllSpans(spans)
-                          .build()));
+              (library, spans) -> {
+                InstrumentationLibrarySpans.Builder spansBuilder =
```
Comment (kubawach) [inline exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/SpanAdapter.java:79]: Similar idea here (move builder to a helper method).

File: sdk/common/src/main/java/io/opentelemetry/sdk/common/InstrumentationLibraryInfo.java [hunk 4]
```diff
@@ -31,7 +31,21 @@
...
+    return new AutoValue_InstrumentationLibraryInfo(name, version, null);
+  }
+
+  /**
+   * Creates a new instance of {@link InstrumentationLibraryInfo}.
+   *
+   * @param name name of the instrumentation library (e.g., "io.opentelemetry.contrib.mongodb"),
+   *     must not be null
+   * @param version version of the instrumentation library (e.g., "1.0.0"), might be null
+   * @return the new instance
```
Comment (kubawach) [inline sdk/common/src/main/java/io/opentelemetry/sdk/common/InstrumentationLibraryInfo.java:43]: add @ param for schemaUrl

File: sdk/common/src/main/java/io/opentelemetry/sdk/internal/ComponentRegistry.java [hunk 5]
```diff
@@ -49,8 +49,24 @@ public final V get(String instrumentationName) {
    * @return the registered value associated with this name and version.
    */
   public final V get(String instrumentationName, @Nullable String instrumentationVersion) {
```
Comment (kubawach) [inline sdk/common/src/main/java/io/opentelemetry/sdk/internal/ComponentRegistry.java:51]: Add to javadoc that "schemUrl" will be set to null?

**Review by kubawach — APPROVED** [review]

File: api/metrics/src/main/java/io/opentelemetry/api/metrics/MeterProvider.java [hunk 6]
```diff
@@ -42,4 +42,19 @@ static MeterProvider noop() {
...
+   * the OpenTelemetry schema.
+   *
+   * @param instrumentationName The name of the instrumentation library, not the name of the
+   *     instrument*ed* library.
+   * @param instrumentationVersion The version of the instrumentation library.
+   * @param schemaUrl The URL of the OpenTelemetry schema being used by this instrumentation
+   *     library.
+   * @return a tracer instance.
+   */
+  default Meter get(String instrumentationName, String instrumentationVersion, String schemaUrl) {
```
Comment (anuraaga) [inline api/metrics/src/main/java/io/opentelemetry/api/metrics/MeterProvider.java:57]: Do we need to make version nullable to allow passing schema without version?

Comment (jkwatson) [reply api/metrics/src/main/java/io/opentelemetry/api/metrics/MeterProvider.java:57]: Yes. Will do

**Review by anuraaga — APPROVED** [review]

**Comment by jkwatson** [comment]
closing, as we're going with the builder.

**Pull request closed without merging**